### PR TITLE
feat(service-registry artifact): allow loading artifact from remote URL

### PR DESCRIPTION
**Pull request by rkpattnaik780**
It should be possible to create/update artifacts using remote URLs.

### Verification Steps
1. Pass url as value for `--file` while creating an artifact.
```
rhoas service-registry artifact create --type OPENAPI --file "https://raw.githubusercontent.com/bolcom/avro-schema-viewer/master/docs/assets/avsc/1.0/schema.avsc" --artifact-id foo_id
```
2. Pass url as value for `--file` to update an artifact.
```
rhoas service-registry artifact update --artifact-id foo_id --group DEFAULT --file "https://raw.githubusercontent.com/bolcom/avro-schema-viewer/master/docs/assets/avsc/1.0/schema.avsc"
```

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### Checklist

- [ ] Documentation added for the feature
- [X] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

**Comment by craicoverflow** [comment]
I don't see any reviewer for this or #1344 s not sure whether you would like them to be reviewed or not?

**Review by craicoverflow — COMMENTED** [review]
```shell
❯ ./rhoas service-registry artifact update --artifact-id foo_id --group DEFAULT --file "https://raw.githubusercontent.com/bolcom/avro-schema-viewer/master/docs/assets/avsc/1.0/schema.avsc"
Loading file from url: https://raw.githubusercontent.com/bolcom/avro-schema-viewer/master/docs/assets/avsc/1.0/schema.avsc
❌ 404 Not Found. Run the command in verbose mode using the -v flag to see more information
```

This makes it seem like the file is not found, but in actual fact it is because the artifact does not exist. The user will not know that.

Overall this works really well 👍🏻  Not approving as I think there improvements that can be made but not outright rejecting it either. If you feel you want things kept as s that is fine, just re-request my review and let me know :)

File: pkg/cmdutil/cmdutil.go
Comment (wtrocki) [inline pkg/cmdutil/cmdutil.go:54]: FYI We already have some helper that has similar content: CreateFileFromStdin.

File: pkg/cmdutil/cmdutil.go
Comment (wtrocki) [inline pkg/cmdutil/cmdutil.go:83]: Not sure if we should be actually delete that file

File: pkg/cmdutil/cmdutil.go
Comment (wtrocki) [reply pkg/cmdutil/cmdutil.go:54]: https://github.com/redhat-developer/app-services-cli/blob/analitics/pkg/cmd/registry/artifact/util/files.go#L12:L13

File: pkg/cmdutil/cmdutil.go
Comment (wtrocki) [inline pkg/cmdutil/cmdutil.go:91]: Logic here works but it is opening the same file twice.
It is better to 1) write content into that file. Set buffer to beginning 
```
 _, err = (*specifiedFile).Write(data)
	if err != nil {
		return nil, err
	}
	_, err = (*specifiedFile).Seek(0, io.SeekStart)
	if err != nil {
		return nil, err
	}
```

**Comment by wtrocki** [comment]
It works

File: pkg/cmdutil/cmdutil.go
Comment (rkpattnaik780) [reply pkg/cmdutil/cmdutil.go]: Should we use a struct for all the utility methods to pass common things like localizer, context, it seems bit odd to have an additional argument for everything?

**Review by wtrocki — APPROVED** [review]

**Comment by craicoverflow** [comment]
This should be merged right?

**Comment by rkpattnaik780** [comment]
> This should be merged right?

Will merge once approved.

**Comment by craicoverflow** [comment]
> > This should be merged right?
> 
> Will merge once approved.

I see that @wtrocki has approved. Do PRs need 2 approvals?

File: pkg/cmd/registry/artifact/util/files.go
Comment (craicoverflow) [inline pkg/cmd/registry/artifact/util/files.go:1]: I know it pre-exists this PR, but going forward we should not call packages "util", and we should update it when we see them.

File: pkg/cmd/registry/artifact/util/files.go
Comment (craicoverflow) [inline pkg/cmd/registry/artifact/util/files.go:55]: i18n?

File: pkg/cmd/registry/artifact/util/files.go
Comment (craicoverflow) [inline pkg/cmd/registry/artifact/util/files.go:67]: i18n?

**Review by craicoverflow — APPROVED** [review]

File: pkg/cmdutil/cmdutil.go
Comment (craicoverflow) [reply pkg/cmdutil/cmdutil.go]: I agree, it also means we must continuously "break" APIs when new requirements are added such as i18n, which may not exist in a util but then if we need to add it we automatically break it..

A better thing to do might be to return a special error type like `InitializeTemporaryFileErr` and in the usage, interpret that and use i18n to create the message outside of the util..

Neither option seems ideal to me. Maybe merge it as it as its such a minor error, there's no point sacrificing usability.

**Pull request merged**
